src/prefect/infrastructure/kubernetes.py

```diff
@@ -263,28 +264,45 @@ async def run(
         if not self.command:
             raise ValueError("Kubernetes job cannot be run with empty command.")
 
-        # if a k8s cluster block is provided to the flow runner, use that
-        if self.cluster_config:
-            self.cluster_config.configure_client()
-        else:
-            # If no block specified, try to load Kubernetes configuration within a cluster. If that doesn't
-            # work, try to load the configuration from the local environment, allowing
-            # any further ConfigExceptions to bubble up.
-            try:
-                kubernetes.config.load_incluster_config()
-            except kubernetes.config.ConfigException:
-                kubernetes.config.load_kube_config()
-
+        self._set_config()
         manifest = self.build_job()
         job_name = await run_sync_in_worker_thread(self._create_job, manifest)
 
+        job_pid = self._get_infrastructure_pid(job_name)
         # Indicate that the job has started
         if task_status is not None:
-            task_status.started(job_name)
+            task_status.started(job_pid)
 
         # Monitor the job
         return await run_sync_in_worker_thread(self._watch_job, job_name)
 
+    async def kill(self, infrastructure_pid: str, grace_seconds: int):
+        self._set_config()
+        job_cluster, job_name = self._parse_infrastructure_pid(infrastructure_pid)
+        current_cluster = self._get_cluster_name()
+        if job_cluster != current_cluster:
+            raise InfrastructureNotAvailable(
+                f"Unable to stop job {job_name!r}: the job is running on cluster ",
+                f"{job_cluster!r}, but your current context is attached to cluster ",
+                f"{current_cluster!r}.",
+            )
+
+        with self.get_batch_client() as batch_client:
+            try:
+                batch_client.delete_namespaced_job(
+                    name=job_name,
+                    namespace=self.namespace,
+                    grace_period_seconds=grace_seconds,
+                    propagation_policy="Foreground",
+                )
+            except kubernetes.client.exceptions.ApiException as exc:
+                if exc.status == 404:
+                    raise InfrastructureNotFound(
+                        f"Unable to stop job {job_name!r}: The job was not found."
+                    )
```

Comment:
With `from exc` this will raise chained. Without, it'll raise as "an exception occurred while handling your exception" eek! :) 

```suggestion
                    raise InfrastructureNotFound(
                        f"Unable to stop job {job_name!r}: The job was not found."
                    ) from exc
```